modules/phase_field/src/ics/MultiSmoothSuperellipsoidIC.C

```diff
@@ -156,13 +173,12 @@ MultiSmoothSuperellipsoidIC::computeSuperellipsoidCenters()
           rr = tmp_rr;
       }
 
-      if (i == 0) rr = _range.norm();
+      if (i == start) rr = _range.norm();
 
       //Check each new center candidate against all accepted ellisoids unless this is the first candidate
-      if (_prevent_overlap == 1 && rr > _bubspac[_gk] && _centers.size() > 0)
+      if (_prevent_overlap && rr > _bubspac[_gk] && _centers.size() > 0)
```

Comment:
You cannot check `_centers.size()` because that vector has been explicitly resized already! You'd have to check `i > 0`, but I think this entire logic is convoluted and overcomplicated.